graphql-dgs-codegen-core/src/test/kotlin/com/netflix/graphql/dgs/codegen/CodeGenTest.kt

```diff
@@ -5065,4 +5065,34 @@ It takes a title and such.
         ).generate()
         assertThat(result.javaDataTypes[0].typeSpec.fieldSpecs[1].type.toString() == "java.lang.String")
     }
+
+    @Test
+    fun `Codegen should generate schema with unsigned int`() {
+        val schema = """
+            scalar UnsignedInt
+            
+            type Continent {
+                code: ID!
+                name: String!
+                population: UnsignedInt
+            }
+            
+            input ContinentInput {
+                numberPopulation: UnsignedInt = 0
+            }
+        """.trimIndent()
+
+        val codeGenResult = CodeGen(
+            CodeGenConfig(
+                schemas = setOf(schema),
+                packageName = basePackageName,
+                typeMapping = mapOf("UnsignedInt" to "java.lang.Long")
+            )
+        ).generate()
+        val dataFetchers = codeGenResult.javaDataFetchers
+        val dataTypes = codeGenResult.javaDataTypes
+        assertThat(dataFetchers).isEmpty()
+        assertThat(dataTypes.size).isEqualTo(2)
```

Comment:
would it make sense to explicitly check the type of the field generated?

Reply:
yes, added explicit asserts for both type and value, thanks!